Implement altering column with `SET DEFAULT` and `DROP DEFAULT`

JanJakes opened the pull request:
This PR implements the support for the following MySQL statements:

```sql
ALTER TABLE _tmp_table CHANGE name SET DEFAULT 'abc';
ALTER TABLE _tmp_table CHANGE name DROP DEFAULT;
```

It required a small refactor so that the CHANGE COLUMN logic can be extracted to a reusable method.

brandonpayton commented on wp-includes/sqlite/class-wp-sqlite-translator.php:
```diff
@@ -3094,8 +3109,21 @@ private function execute_alter() {
+					// While CREATE TABLE normalizes column names to strings, it's better to handle all
+					// variants to handle table definitions with names stored in any valid form.
+					$is_column_name = WP_SQLite_Token::TYPE_STRING === $token->type
+						|| WP_SQLite_Token::TYPE_SYMBOL === $token->type
```
@JanJakes, I think this change should probably be in a separate PR.

AFAICT, it is unrelated to the main purpose of this PR, and a separate PR would allow us to review the codebase and address other places where we could treat names more thoroughly. Maybe there could be a new helper method to check correctly wherever it is needed.

brandonpayton commented on wp-includes/sqlite/class-wp-sqlite-translator.php:
```diff
@@ -3053,16 +3053,31 @@ private function execute_alter() {
-					$this->table_name,
-					$new_field->name,
-					$new_field->mysql_data_type
+				$raw_from_name = 'COLUMN' === $op_subject ? $this->rewriter->skip()->token : $op_raw_subject;
```
Is `ALTER TABLE <table-name> CHANGE COLUMN <column-name> SET DEFAULT <value>` valid MySQL syntax?

Based on the docs [here](https://dev.mysql.com/doc/refman/8.4/en/alter-table.html), it looks like the syntax should be:
```sql
ALTER TABLE table_name ALTER COLUMN column_name SET DEFAULT value;
```

And this seems to be correct in some testing:
```
MariaDB [149697504]> ALTER TABLE bptest CHANGE COLUMN stuff SET DEFAULT 777;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'SET DEFAULT 777' at line 1
MariaDB [149697504]> ALTER TABLE bptest ALTER COLUMN stuff SET DEFAULT 777;
Query OK, 0 rows affected (0.002 sec)
Records: 0  Duplicates: 0  Warnings: 0
```

JanJakes committed:
Update CHANGE COLUMN logic to execute a callback for each column

The CHANGE COLUMN logic can be extracted to a reusable method now. I'm only keeping
the changes in a separate commit for a more useful diff.

The update column callback can be used to modify any columns. Since it is also provided with
the old column definition, it will enable column-modifying statements withtout a full column
definition (like SET/DROP default). In the future, it should also unlock implementing expressions
like MODIFY COLUMN and DROP COLUMN and enable modifying multiple columns at once
to avoid unnecessary table copying.

JanJakes committed:
Extract CHANGE COLUMN logic to a private method

This completes the changes in the previous commit.

JanJakes committed:
Implement altering column with SET DEFAULT and DROP DEFAULT

adamziel commented:
Superseded by #179
